pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2099,6 +2119,35 @@ func isSubnetPublic(rt []*ec2.RouteTable, subnetID string) (bool, error) {
 	return false, nil
 }
 
+func getListener(port api.ServicePort, annotations map[string]string) (*elb.Listener, error) {
```

Comment:
Nit: maybe call this buildListener?  That's more what it is doing, right?  (A comment might be helpful too!)